pkg/manager/member/tikv_scaler.go

```diff
@@ -103,14 +108,30 @@ func (tsd *tikvScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, new
 		return fmt.Errorf("tikvScaler.ScaleIn: failed to get pods %s for cluster %s/%s, error: %s", podName, ns, tcName, err)
 	}
 
-	if tsd.deps.CLIConfig.PodWebhookEnabled {
-		setReplicasAndDeleteSlots(newSet, replicas, deleteSlots)
+	tc, _ := meta.(*v1alpha1.TidbCluster)
+	upNumber := 0
+	storeState := ""
+	for _, store := range tc.Status.TiKV.Stores {
+		if store.State == v1alpha1.TiKVStateUp {
+			upNumber++
+		}
+		if store.PodName == podName {
+			storeState = store.State
+		}
+	}
+	if upNumber < MinimalUpStateOfStores {
```

Comment:
I think it's better to use the replica configuration instead of a hardcode constant `MinimalUpStateOfStores`. The replica configuration can be found via the PD client, see: https://docs.pingcap.com/tidb/stable/pd-configuration-file#replication.

Reply:
Good point.

Reply:
@cvvz You can use `GetConfig` function in the `pkg/pdapi/pdapi.go` to get the config from PD.